homeassistant/components/binary_sensor/insteon_plm.py

```diff
@@ -56,32 +61,58 @@ def should_poll(self):
     @property
     def address(self):
         """Return the address of the node."""
-        return self._address
+        return self._device.address.human
 
     @property
-    def name(self):
+    def id(self):
         """Return the name of the node."""
-        return self._name
+        return self._device.id
 
     @property
-    def is_on(self):
-        """Return the boolean response if the node is on."""
-        sensorstate = self._plm.get_device_attr(self._address, 'sensorstate')
-        _LOGGER.info("Sensor state for %s is %s", self._address, sensorstate)
-        return bool(sensorstate)
+    def name(self):
+        """Return the name of the node. (used for Entity_ID)"""
+        if self._newnames:
+            return self._device.id + '_' + self._state.name
+        else:
+            if self._state.group == 0x01:
+                return self._device.id
+            else:
+                return self._device.id+'_'+str(self._state.group)
 
     @property
     def device_state_attributes(self):
         """Provide attributes for display on device card."""
         insteon_plm = get_component('insteon_plm')
-        return insteon_plm.common_attributes(self)
-
-    def get_attr(self, key):
-        """Return specified attribute for this device."""
-        return self._plm.get_device_attr(self.address, key)
+        return insteon_plm.common_attributes(self._device, self._state)
 
     @callback
-    def async_binarysensor_update(self, message):
+    def async_binarysensor_update(self, deviceid, statename, val):
         """Receive notification from transport that new data exists."""
-        _LOGGER.info("Received update callback from PLM for %s", self._address)
+        _LOGGER.info('Received update calback from PLM for device %s state %s',
+                     deviceid, statename)
         self._hass.async_add_job(self.async_update_ha_state())
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._sensor_type
+
+    @property
+    def is_on(self):
+        """Return the boolean response if the node is on."""
+        sensorstate = self._state.value
+        _LOGGER.info("Sensor for device %s state %s is %s",
+                     self.id, self._state.name, sensorstate)
+        return bool(sensorstate)
+
+    def _stateName_to_sensor_type(self, stateName):
```

Comment:
This could be replaced by a dictionary and then using `TYPES.get(stateName, "")`. Also, should it fail if a state name is not supported?

Reply:
Thank you. Yes, that is a good approach. It should not fail if the stateName is not found. If it is not found, the value of the sensor should just appear as a 1 or a 0, which is the default behavior of a binary_sensor, rather than the images provided when the device_class is not defined. However, after looking at your comment more I see that the default should be `None` anyway.